### DISCUSSION TITLE
Compliance compatible - How much info, if any, does the tool collect?

### DISCUSSION THREAD

**Discussion by billkouts** [renovatebot/renovate#8036] [Request Help] [answered]
Hello!
I am working in a regulated business and I am thinking of tools I can use to improve our engineering workflows. Renovate bot seems like its solving a big problem, so really happy I found it!
But, i need to know what organisation data if any are collected.
I can see the GitHub app requires some write access to the repositories, which makes sense.
I couldn't find anywhere in the docs/website a clear statement about the data that are collected if any.
Can someone help me or point me to the right direction?

I assume, running the tool on prem/self-hosted is one solution. But, that always means an extra cost, so would rather avoid it if possible.

Thank you!

**Comment by rarkins** [comment, marked as answer]
WhiteSource's privacy policy for the hosted app is here: https://www.whitesourcesoftware.com/free-developer-tools/terms-of-use#renovate

The app doesn't keep or cache any repo data between runs, but the backend database does have a list of dependencies and versions per repo plus basic into about any PRs it's created.

If any concerns, please go the self-hosted route - you can probably run it on a $5/month droplet for example.

---

Edit from @HonkingGoose, we now also have [Renovate docs, Security and Permissions](https://docs.renovatebot.com/security-and-permissions/) that explains more about certification, permissions and privacy.

**Reply by billkouts** [reply]
Thank you for the reply!

**Reply by ctessier** [reply]
Hi @rarkins,

I am also interested in this topic for a client, but I struggle to understand how the self-hosted alternative works. 

Does it checkout the source code for the run on the host somehow, or does it use the read access to the Github repo to only browse the files and read the deps files remotely (like package.json or composer.json)?

Also, can you confirm that with the self-hosted way, sensitive information like source code and Github token for instance only transit between the host and Github, and nothing is sent to a Renovate server of some sort?

Thank you very much.
Have a good day.

**Reply by viceice** [reply]
Renovate will do a local git checkout which will be deleted after run by default.

The self-hosted runner doesn't send any information to renovate owners.

**Reply by rarkins** [reply]
Neither the fully open source CLI or WhiteSource Renovate On-Premises edition send any repository or dependency data home or utilise any hosted service we run.

**Reply by tguvdamm** [reply]
> Neither the fully open source CLI or WhiteSource Renovate On-Premises edition send any repository or dependency data home or utilise any hosted service we run.


@rarkins can you confirm this is still the case? Is this listed in the Terms somewhere?

**Reply by rarkins** [reply]
It's still the case. There's no such thing as terms for OSS so your question is invalid

**Reply by tguvdamm** [reply]
Thanks!
